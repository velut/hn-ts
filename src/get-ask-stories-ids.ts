import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getAskStoriesIds` returns the list of up to 200 IDs of the "Ask HN" stories.
 *
 * @see {@link https://github.com/HackerNews/API#ask-show-and-job-stories}
 */
export async function getAskStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/askstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
