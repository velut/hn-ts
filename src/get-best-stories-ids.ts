import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getBestStoriesIds` returns the list of up to 500 IDs of the best HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#new-top-and-best-stories}
 */
export async function getBestStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/beststories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
