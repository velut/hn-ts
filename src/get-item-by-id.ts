import { getRawItemById } from "./get-raw-item-by-id";
import { Item } from "./item";
import { normalizeRawItem } from "./normalize-raw-item";

/**
 * `getItemById` returns the `Item` with the given ID.
 * If the item is not available, `getItemById` returns `undefined`.
 *
 * @see {@link Item}
 * @see {@link https://github.com/HackerNews/API#items}
 */
export async function getItemById({
  id,
}: {
  id: number;
}): Promise<Item | undefined> {
  const rawItem = await getRawItemById({ id });
  if (!rawItem) {
    return undefined;
  }

  return normalizeRawItem({ rawItem });
}
