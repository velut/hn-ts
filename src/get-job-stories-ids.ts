import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getJobStoriesIds` returns the list of up to 200 IDs of the job HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#ask-show-and-job-stories}
 */
export async function getJobStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/jobstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
