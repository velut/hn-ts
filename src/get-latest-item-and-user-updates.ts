import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";
import { Updated } from "./updated";

/**
 * `getLatestItemAndUserUpdates` returns the latest updates for items and users.
 *
 * @see {@link https://github.com/HackerNews/API#changed-items-and-profiles}
 */
export async function getLatestItemAndUserUpdates(): Promise<Updated> {
  const endpoint = `${hnApiBaseUrl}/updates.json`;
  const res = await fetch(endpoint);
  const updates = await res.json();
  return updates;
}
