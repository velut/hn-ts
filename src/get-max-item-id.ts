import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getMaxItemId` returns the ID of the most recent item available from the API.
 *
 * @see {@link https://github.com/HackerNews/API#max-item-id}
 */
export async function getMaxItemId(): Promise<number> {
  const endpoint = `${hnApiBaseUrl}/maxitem.json`;
  const res = await fetch(endpoint);
  const id = await res.json();
  return id;
}
