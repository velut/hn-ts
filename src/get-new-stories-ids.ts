import fetch from "isomorphic-unfetch";
import { hnApiBaseUrl } from "./hn-api-base-url";

/**
 * `getNewStoriesIds` returns the list of up to 500 IDs of the newer HN stories.
 *
 * @see {@link https://github.com/HackerNews/API#new-top-and-best-stories}
 */
export async function getNewStoriesIds(): Promise<number[]> {
  const endpoint = `${hnApiBaseUrl}/newstories.json`;
  const res = await fetch(endpoint);
  const ids = await res.json();
  return ids;
}
