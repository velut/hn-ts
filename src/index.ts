/**
 * This package TODO:
 *
 * @packageDocumentation
 */

export * from "./get-ask-stories-ids";
export * from "./get-best-stories-ids";
export * from "./get-item-by-id";
export * from "./get-job-stories-ids";
export * from "./get-max-item-id";
export * from "./get-new-stories-ids";
export * from "./get-raw-item-by-id";
export * from "./get-raw-user-by-id";
export * from "./get-show-stories-ids";
export * from "./get-top-stories-ids";
export * from "./get-user-by-id";
export * from "./hn-api-base-url";
export * from "./item";
export * from "./item-type";
export * from "./raw-item";
export * from "./raw-item-type";
export * from "./raw-user";
export * from "./user";
