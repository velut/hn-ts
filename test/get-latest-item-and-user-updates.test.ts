/** @jest-environment setup-polly-jest/jest-environment-node */
// See https://netflix.github.io/pollyjs/#/test-frameworks/jest-jasmine?id=supported-test-runners

import NodeHttpAdapter from "@pollyjs/adapter-node-http";
import FSPersister from "@pollyjs/persister-fs";
import * as path from "path";
import { setupPolly } from "setup-polly-jest";
import { getLatestItemAndUserUpdates } from "../src";

describe("getLatestItemAndUserUpdates", () => {
  setupPolly({
    adapters: [NodeHttpAdapter],
    persister: FSPersister,
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, "./__recordings__"),
      },
    },
    recordFailedRequests: true,
  });

  it("returns the latest updates", async () => {
    expect.assertions(3);

    const updated = await getLatestItemAndUserUpdates();
    const { items, profiles } = updated;

    expect(items.length).toBeGreaterThanOrEqual(1);
    expect(profiles.length).toBeGreaterThanOrEqual(1);
    expect(updated).toMatchSnapshot();
  });
});
