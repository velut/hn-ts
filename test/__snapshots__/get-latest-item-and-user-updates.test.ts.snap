// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLatestItemAndUserUpdates returns the latest updates 1`] = `
{
  "items": [
    35794474,
    35791125,
    35794484,
    35790166,
    35794595,
    35794462,
    35794506,
    18160958,
    35794597,
    35792264,
    35792868,
    35794593,
    35790217,
    35791189,
    35793764,
    35793828,
    35792691,
    35787484,
    35792808,
    35787684,
    35790618,
    35793975,
    35792677,
    35787855,
    35791005,
    35794330,
    35793456,
    35792910,
    35791562,
    35788808,
    35794381,
    35792141,
    35794111,
    35792741,
    35781609,
    35793836,
    35793744,
    35792776,
    35792078,
    35790996,
    35792734,
    35790791,
    35792196,
    35794594,
    35794065,
    35794103,
    35788930,
    35792672,
    35794570,
    35791315,
    35794596,
    35788353,
    35779034,
    35784814,
    35786820,
    35787080,
    35779796,
    35785231,
    35787287,
    35741479,
    35794505,
    35789730,
    35790647,
    35790077,
    35787454,
    35790367,
    18160618,
    35792360,
    35789963,
    35785319,
    35791575,
    35791947,
    35788120,
    35781605,
    35787707,
    35785255,
    35762682,
    35786086,
    35793009,
    35794416,
    35794002,
    35790092,
    35778735,
    35793187,
  ],
  "profiles": [
    "jofer",
    "adrian_mrd",
    "mikrl",
    "quantgenius",
    "MD3XTER",
    "whywhywhywhy",
    "dylan604",
    "reiichiroh",
    "acomjean",
    "kiawe_fire",
    "tjr",
    "sleepybrett",
    "chrislattner",
    "DiggyJohnson",
    "officeplant",
    "acchow",
    "DyslexicAtheist",
    "lampshades",
    "pcurve",
    "mikestew",
    "ZeroGravitas",
    "yunyu",
    "hbn",
    "lairv",
    "Veen",
    "Someone",
  ],
}
`;
